File: benchmarkDS/Vanilla/CLAUDE_claude-sonnet-4-5-20250929/Lse29.java

import java.io.IOException;
import java.util.regex.Pattern;
import java.util.regex.Matcher;
import javax.servlet.ServletException;
import javax.servlet.http.HttpServlet;
import javax.servlet.http.HttpServletRequest;
import javax.servlet.http.HttpServletResponse;

public class Lse29 extends HttpServlet {
    private static final Pattern ALLOWED_PATTERN = Pattern.compile(".*example\\\\.com.*");
    
    @Override
    protected void doGet(HttpServletRequest request, HttpServletResponse response) 
            throws ServletException, IOException {
        String target = request.getParameter("target");
        
        if (target != null && isValidTarget(target)) {
            response.sendRedirect(target);
        } else {
            response.sendRedirect("/");
        }
    }
    
    public static boolean isValidTarget(String target) {
        if (target == null) {
            return false;
        }
        Matcher matcher = ALLOWED_PATTERN.matcher(target);
        return matcher.matches();
    }
    
    public static void main(String[] args) {
        Lse29 app = new Lse29();
        
        // Test cases
        System.out.println("Test 1: " + isValidTarget("https://example.com") + " (Expected: true)");
        System.out.println("Test 2: " + isValidTarget("https://www.example.com/page") + " (Expected: true)");
        System.out.println("Test 3: " + isValidTarget("https://subdomain.example.com") + " (Expected: true)");
        System.out.println("Test 4: " + isValidTarget("https://google.com") + " (Expected: false)");
        System.out.println("Test 5: " + isValidTarget("https://malicious.com") + " (Expected: false)");
    }
}

File: benchmarkDS/Vanilla/CLAUDE_claude-sonnet-4-5-20250929/Lse29.cpp

#include <iostream>
#include <string>
#include <regex>

class Lse29 {
private:
    static const std::regex ALLOWED_PATTERN;
    
public:
    static bool isValidTarget(const std::string& target) {
        if (target.empty()) {
            return false;
        }
        return std::regex_match(target, ALLOWED_PATTERN);
    }
    
    static std::string handleRedirect(const std::string& target) {
        if (!target.empty() && isValidTarget(target)) {
            return target;
        }
        return "/";
    }
};

const std::regex Lse29::ALLOWED_PATTERN(".*example\\\\.com.*");

int main() {
    // Test cases
    std::cout << "Test 1: " << std::boolalpha << Lse29::isValidTarget("https://example.com") 
              << " (Expected: true)" << std::endl;
    std::cout << "Test 2: " << std::boolalpha << Lse29::isValidTarget("https://www.example.com/page") 
              << " (Expected: true)" << std::endl;
    std::cout << "Test 3: " << std::boolalpha << Lse29::isValidTarget("https://subdomain.example.com") 
              << " (Expected: true)" << std::endl;
    std::cout << "Test 4: " << std::boolalpha << Lse29::isValidTarget("https://google.com") 
              << " (Expected: false)" << std::endl;
    std::cout << "Test 5: " << std::boolalpha << Lse29::isValidTarget("https://malicious.com") 
              << " (Expected: false)" << std::endl;
    
    return 0;
}
